libraries/Microsoft.Bot.Builder.Core.State/IStateStorageEntry.cs

```diff
@@ -0,0 +1,13 @@
+﻿namespace Microsoft.Bot.Builder.Core.State
+{
+    public interface IStateStorageEntry
+    {
+        string Namespace { get; }
+
+        string Key { get;  }
+
+        T GetValue<T>() where T : class, new();
+
+        void SetValue<T>(T value) where T : class;
```

Comment:
>void SetValue<T>(T value) where T : class; [](start = 8, length = 42)

I'm only implementing the getValue() for JS. And I shortened it to get() as that's what a JS developer expects.

Reply:
Yep, catering to the JS patterns here makes sense to me.